src/frontend/next/src/pages/search.tsx

```diff
@@ -0,0 +1,62 @@
+import { FormEvent, useState } from 'react';
+import { makeStyles } from '@material-ui/core/styles';
+import { withRouter, NextRouter } from 'next/router';
+
+import SearchResults from '../components/SearchResults';
+import SearchBar from '../components/SearchBar';
+import BackToTopButton from '../components/BackToTopButton';
+
+const useStyles = makeStyles(() => ({
+  anchor: {
+    position: 'absolute',
+    top: 0,
+  },
+}));
+
+const SearchPage = ({ router }: { router: NextRouter }) => {
+  const classes = useStyles();
+  // We synchronize the `text` and `filter` values to the URL's query string
+  // Router query object for a query can be an array if url becomes text=123&text=456
+  // https://stackoverflow.com/questions/60110364/type-string-string-is-not-assignable-to-type-string
```

Comment:
the `text=...&text=...` case is a bug, surely.  If someone does provide multiple values for the same param, what are we supposed to do?  No matter which one we choose, we might be wrong.  I think it's reasonable to error in this case instead.  Get the user to fix their URL.

Reply:
A type issue was popping up, the since the returned values from `router.query.text` and `router.query.filter` can either be `string` or `string[]`

Reply:
Alright, let's do this for now and fix later.